### PR TITLE
 [invite updates] feature/CAES-209: implemented deferred invite updates

### PR DESCRIPTION
**Pull request by ipopov4xxi**

### CODE REVIEW

File: src/Entity/Item.php [hunk 1]
```diff
@@ -101,6 +101,13 @@ class Item
      */
     protected $access;
 
+    /**
+     * @var ItemUpdate|null
+     *
+     * @ORM\OneToOne(targetEntity="App\Entity\ItemUpdate", mappedBy="item", orphanRemoval=true, cascade={"PERSIST"})
```
Comment (aburov-4xxi) [inline src/Entity/Item.php]: PERSIST - persist, use lower case

File: src/Entity/ItemUpdate.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+     *
+     * @ORM\Id
+     * @ORM\Column(type="uuid", unique=true)
+     */
+    protected $id;
+
+    /**
+     * @var string|null
+     *
+     * @ORM\Column(type="text", nullable=false)
```
Comment (aburov-4xxi) [inline src/Entity/ItemUpdate.php]: nullable=false is default

File: src/Entity/ItemUpdate.php [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    /**
+     * @var UuidInterface
+     *
+     * @ORM\Id
+     * @ORM\Column(type="uuid", unique=true)
+     */
+    protected $id;
+
+    /**
+     * @var string|null
```
Comment (aburov-4xxi) [inline src/Entity/ItemUpdate.php]: not null

File: src/Entity/ItemUpdate.php [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+     *
+     * @ORM\Column(type="datetime")
+     */
+    protected $lastUpdated;
+
+    public function __construct(Item $item, User $user)
+    {
+        $this->id = Uuid::uuid4();
+        $this->updatedBy = $user;
+
```
Comment (aburov-4xxi) [inline src/Entity/ItemUpdate.php]: extra spaces

File: src/Factory/View/ItemViewFactory.php [hunk 5]
```diff
@@ -68,4 +71,18 @@ private function getOwnerId(Item $item): string
 
         return $this->userRepository->getByItem($ownerItem)->getId()->toString();
     }
+
+    private function getUpdateView(?ItemUpdate $update)
```
Comment (aburov-4xxi) [inline src/Factory/View/ItemViewFactory.php]: `: ?UpdatedView`

File: src/Services/InviteHandler.php [hunk 6]
```diff
@@ -45,4 +46,46 @@ public function inviteToItem(InviteCollectionRequest $request)
 
         $this->entityManager->flush();
     }
+
+    /**
+     * @param InviteCollectionRequest $request
+     *
+     * @throws \Exception
+     */
+    public function updateInvites(InviteCollectionRequest $request)
```
Comment (aburov-4xxi) [inline src/Services/InviteHandler.php]: : void

File: src/Entity/ItemUpdate.php (hunk 4 above)
Comment (ipopov4xxi) [reply src/Entity/ItemUpdate.php]: To separate internal entity logic and contructor parameters maping.

File: src/Services/InviteHandler.php (hunk 6 above)
Comment (ipopov4xxi) [reply src/Services/InviteHandler.php]: Fixed

File: src/Entity/ItemUpdate.php (hunk 2 above)
Comment (ipopov4xxi) [reply src/Entity/ItemUpdate.php]: Fixed

File: src/Entity/ItemUpdate.php (hunk 3 above)
Comment (ipopov4xxi) [reply src/Entity/ItemUpdate.php]: Fixed

File: src/Entity/Item.php (hunk 1 above)
Comment (ipopov4xxi) [reply src/Entity/Item.php]: Fixed

**Pull request merged**